### PR TITLE
 Adding unit tests for multiply and add [Issue 7]

### PR DESCRIPTION
**Pull request by tomichaw**
#7 

Unit test for multiply fails:
./modulo_lib_test
   modulo_lib_test.cc:40 Expectation not met: 0 == 1
modulo_multiply: FAIL
modulo_add: PASS
Some test cases failed!

Added bug #8 to investigate this further.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
 modulo_lib.o: modulo_lib.cc modulo_lib.h 
-	g++ -c $< -o $@
+	g++ -c modulo_lib.cc -o $@
 
 modulo_lib_test.o: modulo_lib_test.cc modulo_lib.h 
-	g++ -c $< -o $@
+	g++ -c modulo_lib_test.cc -o $@
+
+modulo_lib_test: modulo_lib.o modulo_lib_test.o
+	g++ modulo_lib.o modulo_lib_test.o -o $@
```
Comment (birukw) [inline Makefile]: Here, you can use the automatic make variable $^ to refer to all prerequisites :)

File: modulo_lib_test.cc [hunk 2]
```diff
@@ -27,13 +27,29 @@ std::string status(bool passed) {
 }
 
 
+struct TestInputs {
+	int a, b, mod;
+};
+
 bool test_modulo_multiply() {
-    // TODO(tomichaw): Implement this.
+	TestInputs inputs[] = { {1, 2, 3}, {2, 2, 3}, {2, 2, 4} };
```
Comment (birukw) [inline modulo_lib_test.cc]: It would be more readable if you put each input in its own line. Same for below.

**Review by birukw — APPROVED** [review]
Maybe it is time we start looking into continuous integration.

File: modulo_lib_test.cc (hunk 2 above)
Comment (birukw) [reply modulo_lib_test.cc]: Done.

File: Makefile (hunk 1 above)
Comment (birukw) [reply Makefile]: Cool! Done.

Comment (tomichaw) [reply Makefile]: Cool! Done.

File: modulo_lib_test.cc (hunk 2 above)
Comment (tomichaw) [reply modulo_lib_test.cc]: Done.

**Review by tomichaw — COMMENTED** [review]
I agree about CI.

**Pull request merged**